docs/DeveloperGuide.md

```diff
@@ -1,33 +1,372 @@
+If the item is in `Lecture`, `Lab`, `Tutorial`, or `Exam` type, there will be a check in the function `isValid(command)` to ensure the module code included in the item is valid.
+If the item is in `Lecture`, `Lab`, or `Tutorial`type, it can be added as recurring items,
+while only one `Todo`, `Deadline`, `Exam` or `Activity` item can be added at a time.   
+
+### Mark a task as done feature
+This feature is facilitated by `DoneCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/done_command_SD.JPG" alt="" width="750"/><br/>
+Note: It first extracts the task number from the user input prior to `convertTaskNumberToCalendarNumber`. 
+There will be a check in the function `markTaskAsDone(calendarNumber)` to ensure that the calendar item being marked as done is a task. 
+
+
+### Additional information of an event feature
+This feature is facilitated by `AddInfoCommand` and the `ViewInfoCommand`.
+The following sequence diagram show how the `execute()` operation of `AddInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/addInfoCommand_SD.JPG" alt="" width="750"/><br/>
+The following sequence diagram show how the `execute()` operation of `ViewInfoCommand` works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/viewInfoCommand_SD.JPG" alt="" width="750"/><br/>
+
+Note: It first extracts the event number from the user input prior to `convertEventNumberToCalendarNumber`. 
+
+### Delete a calendar item feature
+This feature is facilitated by `DeleteCommand`.
+The following sequence diagram show how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/deleteCommand_SD.JPG" alt="" width="750"/><br/>
+
+Note: It first extracts the task/event number from the user input prior to `convertTaskNumberToCalendarNumber` and `convertEventNumberToCalendarNumber` respectively. 
+
+
+### Find a calendar item feature
+This feature is facilitated by `FindCommand`.
+The following sequence diagram show how the `execute()` operation works when the user searches the entire calendar.:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/findCommand_SD.JPG" alt="" width="500"/><br/>
+
+The search for tasks or events feature has a similar sequence diagram. The difference is the varying condition. Depending
+on whether the user searches for tasks or events, the condition will check for the instance of either the task or event respectively.
+
+### Print tasks feature
+This feature is facilitated by `PrintTasksCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to print the list of tasks stored in the program.
+![print_tasks_command_sd](images/PrintTasksCommand_SD.png)
```

Comment:
I think there's a formatting error on your UML diagrams, but the dotted line needs to extend to the centre of the deletion cross!